### PR TITLE
ddl: add check against _tidb_rowid as column name

### LINKED ISSUE #12835: Duplicate column error when `_tidb_rowid` as column name
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
```sql
create table t (_tidb_rowid int);
select * from t;
```

2. What did you expect to see?
```
Empty set (0.00 sec)
```


3. What did you see instead?
```
ERROR 1060 (42S21): Duplicate column name '_tidb_rowid'
```

4. What version of TiDB are you using (`tidb-server -V` or run `select tidb_version();` on TiDB)?
```sql
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| tidb_version()                                                                                                                                                                                                                                                                                                              |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Release Version: v4.0.0-alpha-561-g6c8aa3d
Git Commit Hash: 6c8aa3da1e2ee27d6bb0a0a9b4a1a2b341f896b2
Git Branch: master
UTC Build Time: 2019-10-20 08:33:44
GoVersion: go version go1.13.3 linux/amd64
Race Enabled: false
TiKV Min Version: v3.0.0-60965b006877ca7234adaced7890d7b029ed1306
Check Table Before Drop: false |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+

```

### PR DESCRIPTION
**Pull request by zhongzc**
### What problem does this PR solve? 
Fix #12835.

### What is changed and how it works?
Check column name when building table info. It forbids creating table contains column `_tidb_rowid`.

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)

Code changes
N/A

Side effects
N/A

Related changes
N/A

Release note
N/A

### CODE REVIEW

**Comment by wjhuang2016** [comment]
Thanks for your contribution. The _tidb_rowid exists only when the table's primary key is not an integer or the table doesn't have a primary key. 
```
mysql> create table t (_tidb_rowid int, primary key (_tidb_rowid));
Query OK, 0 rows affected (0.01 sec)

mysql> select * from t;
Empty set (0.00 sec)
```
We need to discuss how to deal with _tidb_rowid

**Comment by zhongzc** [comment]
@SunRunAway found the problem in this case. @wjhuang2016 
```mysql
mysql> create table t (_tidb_rowid int, primary key (_tidb_rowid));
Query OK, 0 rows affected (0.00 sec)

mysql> select * from t;
Empty set (0.00 sec)

mysql> insert into t values (1);
ERROR 1105 (HY000): insert, update and replace statements for _tidb_rowid are not supported.
```

File: ddl/ddl_api.go [hunk 1]
```diff
@@ -868,6 +868,7 @@ func checkColumnValueConstraint(col *table.Column) error {
 
 func checkDuplicateColumn(cols []interface{}) error {
 	colNames := set.StringSet{}
+	colNames.Insert("_tidb_rowid")
```
Comment (zimulala) [inline ddl/ddl_api.go]: Mysql does not have this behavior. Please add a comment about why this code was added. 
How about the operation of `alter table ... add column _tidb_rowid`?

**Comment by tangenta** [comment]
@zhongzc Please address the comments above. Can you add an error like `ErrUnsupportedReservedWord` in [ddl.go](https://github.com/pingcap/tidb/blob/875873b7ca4267c0bbb2e4fe31c1561dea3f495d/ddl/ddl.go#L143)?

File: ddl/ddl_api.go (hunk 1 above)
Comment (zhongzc) [reply ddl/ddl_api.go]: @zimulala Thanks for pointing out the problem. I will fix it soon.

File: infoschema/infoschema.go [hunk 2]
```diff
@@ -328,16 +330,17 @@ const (
...
-	codeMultiplePriKey   = 1068
-	codeTooManyKeyParts  = 1070
-	codeKeyNameDuplicate = 1061
-	codeKeyNotExists     = 1176
+	codeDatabaseExists         = 1007
+	codeTableExists            = 1050
+	codeBadTable               = 1051
+	codeBadUser                = 3162
+	codeColumnExists           = 1060
+	codeReservedColumnConflict = 1066
```
Comment (tangenta) [inline infoschema/infoschema.go]: According to MySQL [documentation](https://dev.mysql.com/doc/refman/8.0/en/server-error-reference.html), 
> Error number: 1066; Symbol: ER_NONUNIQ_TABLE; SQLSTATE: 42000
Message: Not unique table/alias: '%s'

The code 1000-2000, 3000-4000 is used for MySQL error.
Maybe you should use something TiDB-specific, like 215.

File: ddl/column_test.go [hunk 3]
```diff
@@ -978,7 +978,7 @@ func (s *testColumnSuite) colDefStrToFieldType(c *C, str string) *types.FieldTyp
 func (s *testColumnSuite) TestFieldCase(c *C) {
```
Comment (tangenta) [inline ddl/column_test.go]: Could you please add some tests for `_tidb_rowid`?

**Pull request closed without merging**